src/jit/lower.cpp

```diff
@@ -831,26 +831,44 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     // Struct can be split into register(s) and stack on ARM
     if (info->isSplit)
     {
-        if (arg->OperGet() != GT_OBJ)
+        assert(arg->OperGet() != GT_OBJ || arg->OperGet() != GT_FIELD_LIST);
+        // TODO: Need to check correctness for FastTailCall
+        if (call->IsFastTailCall())
         {
-            NYI_ARM("Lowering: Oper for struct argument is not GT_OBJ");
+            NYI_ARM("lower: struct argument by fast tail call");
         }
 
         putArg = new (comp, GT_PUTARG_SPLIT)
             GenTreePutArgSplit(arg, info->slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(info->numSlots), info->numRegs,
                                info->isHfaRegArg, call->IsFastTailCall(), call);
 
-        // Set GC Pointer info
+        // If struct argument is morphed to GT_FIELD_LIST node(s),
+        // we can know GC info by type of each GT_FIELD_LIST node.
+        // So we skip setting GC Pointer info.
+        //
         GenTreePutArgSplit* argSplit = putArg->AsPutArgSplit();
-        BYTE*               gcLayout = new (comp, CMK_Codegen) BYTE[info->numSlots + info->numRegs];
-        unsigned            numRefs  = comp->info.compCompHnd->getClassGClayout(arg->gtObj.gtClass, gcLayout);
-        argSplit->setGcPointers(numRefs, gcLayout);
+        if (arg->OperGet() != GT_FIELD_LIST)
```

Comment:
I'm curious why the gc info hasn't been initialized on the `GT_OBJ`, but at least you could check `arg->AsObj->IsGCInfoInitialized()` first.